### PR TITLE
[master] 3.3.5 -> master cherry-pick

### PR DESCRIPTION
**Pull request by funjoker**
**Changes proposed:**

-  Cherry picking

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

### CODE REVIEW

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp [hunk 1]
```diff
@@ -716,6 +717,23 @@ class npc_death_ravager : public CreatureScript
...
 };
 
+// 29528 -  Inoculate Nestlewood Owlkin
+class spell_inoculate_nestlewood : public AuraScript
+{
+    PrepareAuraScript(spell_inoculate_nestlewood);
+
+    void PeriodicTick(AuraEffect const* /*aurEff*/)
+    {
+        if (GetTarget()->GetTypeId() != TYPEID_UNIT) // prevent error reports in case ignored player target
```
Comment (matanshukry) [inline src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp:727]: Needed? spell 29528 effect 0 uses `Targets (38, 0) (TARGET_UNIT_NEARBY_ENTRY, NO_TARGET)`,
TARGET_UNIT_NEARBY_ENTRY should only result in TARGET_OBJECT_TYPE_UNIT ?

File: src/server/scripts/Northrend/zone_wintergrasp.cpp [hunk 2]
```diff
@@ -579,6 +580,23 @@ class spell_wintergrasp_defender_teleport_trigger : public SpellScriptLoader
     }
 };
 
+// 58549 Tenacity
+// 59911 Tenacity
```
Comment (matanshukry) [inline src/server/scripts/Northrend/zone_wintergrasp.cpp:584]: `As of November 16, 2010 and the release of Patch 4.0.3, Tenacity no longer applies to players during the battle.`
Seems this spell was only used during the battle, so we can remove this completely.

File: src/server/scripts/Spells/spell_generic.cpp [hunk 3]
```diff
@@ -610,6 +712,29 @@ class spell_gen_cannibalize : public SpellScript
     }
 };
 
+// 45524 Chains of Ice
```
Comment (matanshukry) [inline src/server/scripts/Spells/spell_generic.cpp:715]: 45524 doesn't seem related to this

File: src/server/scripts/Spells/spell_generic.cpp [hunk 4]
```diff
@@ -1837,6 +2006,22 @@ class spell_gen_obsidian_armor : public AuraScript
     }
 };
 
+// 55342 Mirror Image
+class spell_gen_one_tick_dummy : public AuraScript
```
Comment (matanshukry) [inline src/server/scripts/Spells/spell_generic.cpp:2010]: 55342 doesn't have any effect with SPELL_AURA_PERIODIC_DUMMY.
EFFECT_1 is SPELL_EFFECT_DUMMY, so probably changed and can now be removed.

File: src/server/scripts/Spells/spell_generic.cpp [hunk 5]
```diff
@@ -2152,6 +2337,116 @@ class spell_gen_remove_flight_auras : public SpellScript
...
+// 58517 Grievous Wound
+// 59262 Grievous Wound
+class spell_gen_remove_on_full_health : public AuraScript
+{
+    PrepareAuraScript(spell_gen_remove_on_full_health);
+
+    void PeriodicTick(AuraEffect const* aurEff)
+    {
+        // if it has only periodic effect, allow 1 tick
+        bool onlyEffect = !GetSpellInfo()->GetEffect(EFFECT_1)->IsEffect() && !GetSpellInfo()->GetEffect(EFFECT_2)->IsEffect();
```
Comment (matanshukry) [inline src/server/scripts/Spells/spell_generic.cpp:2409]: Will cause a crash.
31956 only has one effect, `GetSpellInfo()->GetEffect(EFFECT_1)` will be nullptr,
hence `nullptr->IsEffect()`.

Possible solutions:
1. `bool onlyEffect = GetSpellInfo()->GetEffects().size() == 1` 
that's assuming spells can't have Effect=0, since now in master it's not an array but rather of structs but vector of pointers. But I'm not sure about that

2. Use a loop and do it before Periodic tick, once. Something like `Load()`

File: src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp (hunk 1 above)
Comment (funjoker) [reply src/server/scripts/Kalimdor/zone_azuremyst_isle.cpp:727]: well we will keep it for now, for the sake of sync with 3.3.5

File: src/server/scripts/Northrend/zone_wintergrasp.cpp (hunk 2 above)
Comment (funjoker) [reply src/server/scripts/Northrend/zone_wintergrasp.cpp:584]: Spell still exists we gonna leave it for now

File: src/server/game/Spells/SpellMgr.cpp [hunk 6]
```diff
@@ -3575,6 +3651,12 @@ void SpellMgr::LoadSpellInfoCorrections()
         spellInfo->RangeEntry = sSpellRangeStore.LookupEntry(13); // 50000yd
     });
 
+    // Summon Corpse Scarabs
+    ApplySpellFix({ 28864, 29105 }, [](SpellInfo* spellInfo)
+    {
+        spellInfo->Effects[EFFECT_0].RadiusEntry = sSpellRadiusStore.LookupEntry(EFFECT_RADIUS_10_YARDS);
```
Comment (illfated) [inline src/server/game/Spells/SpellMgr.cpp:3657]:
```
/home/travis/build/TrinityCore/TrinityCore/src/server/game/Spells/SpellMgr.cpp:3657:20:
 fatal error: no member named 'Effects' in 'SpellInfo'
        spellInfo->Effects[EFFECT_0].RadiusEntry = sSpellRadiusStore.Loo...
        ~~~~~~~~~  ^
```

File: src/server/game/Spells/SpellMgr.cpp [hunk 7]
```diff
@@ -3207,6 +3207,82 @@ void SpellMgr::LoadSpellInfoCorrections()
...
+        }, [](SpellInfo* spellInfo)
+        {
+            // first effect has correct amplitude
+            const_cast<SpellEffectInfo*>(spellInfo->GetEffect(EFFECT_1))->Amplitude = const_cast<SpellEffectInfo*>(spellInfo->GetEffect(EFFECT_0))->Amplitude;
+        });
+
+        // Vomit
+        ApplySpellFix({ 43327 }, [](SpellInfo* spellInfo)
+        {
+            const_cast<SpellEffectInfo*>(spellInfo->GetEffect(EFFECT_1))->Amplitude = 1 * IN_MILLISECONDS;
```
Comment (Shauren) [inline src/server/game/Spells/SpellMgr.cpp:3243]: Should change ApplyAuraPeriod, not Amplitude (3.3.5 has wrongly swapped names)
* applies to all Amplitude changes from this commit, not just line im leaving this comment on

File: src/server/game/Spells/SpellInfo.cpp [hunk 8]
```diff
@@ -3745,13 +3745,19 @@ uint32 SpellInfo::GetMaxTicks() const
                 case SPELL_AURA_PERIODIC_TRIGGER_SPELL:
                 case SPELL_AURA_PERIODIC_TRIGGER_SPELL_WITH_VALUE:
                 case SPELL_AURA_PERIODIC_HEALTH_FUNNEL:
-                    if (effect->ApplyAuraPeriod != 0)
-                        return DotDuration / effect->ApplyAuraPeriod;
+                    // skip infinite periodics
+                    if (effect->Amplitude > 0 && DotDuration > 0)
+                    {
+                        totalTicks = static_cast<uint32>(DotDuration) / effect->Amplitude;
```
Comment (Shauren) [inline src/server/game/Spells/SpellInfo.cpp:3751]: Change back to ApplyAuraPeriod

File: sql/base/auth_database.sql [hunk 9]
```diff
@@ -2399,7 +2399,8 @@ INSERT INTO `updates` VALUES
 ('2020_12_31_00_auth.sql','05C9C105D55C6588CDA0D75AE3B135B7E6B54C06','RELEASED','2020-12-31 12:58:21',0),
 ('2021_01_10_00_auth.sql','4430F5CC9A31DBBEA4E437E980E9F26AC919C016','RELEASED','2021-01-10 12:29:57',0),
 ('2021_01_13_00_auth.sql','F7C15519FFA0FE27EED859343B58714624A302E6','RELEASED','2021-01-13 11:19:52',0),
-('2021_02_09_00_auth.sql','F00ABFF6E3C3F5ACE3444C9D70BADC764C8B3CE2','RELEASED','2021-02-09 17:22:24',0);
+('2021_02_09_00_auth.sql','F00ABFF6E3C3F5ACE3444C9D70BADC764C8B3CE2','RELEASED','2021-02-09 17:22:24',0),
+('2021_03_14_00_auth_2017_12_17_00_auth.sql','2CD99730D4D32DBF0584CD5B1AA6F8F4AE3DA975','RELEASED','2021-03-14 00:00:00',0);
```
Comment (Shauren) [inline sql/base/auth_database.sql:2403]: Auth files must not be renamed when cherry picking, keep original name

**Review by Shauren — CHANGES_REQUESTED** [review]

File: sql/base/auth_database.sql (hunk 9 above)
Comment (funjoker) [reply sql/base/auth_database.sql:2403]: how to ensure they are applied at last? i add prefix like in world

Comment (Shauren) [reply sql/base/auth_database.sql:2403]: order is less important than not applying the update twice (auth db is supposed to be shareable)

Comment (funjoker) [reply sql/base/auth_database.sql:2403]: Ok

**Review by Shauren — APPROVED** [review]
